src/Components/SingleArticle.jsx

```diff
@@ -0,0 +1,52 @@
+import { getSingleArticle } from "../api";
+import { useParams } from "react-router";
+import { useState, useEffect } from "react";
+
+const SingleArticle = () => {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState({});
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    getSingleArticle(article_id)
+      .then((article) => {
+        setArticle(article);
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, [article_id]);
+
+  if (loading) {
+    <p>Loading Article...</p>;
```

Comment:
This needs a return (you might also need to put onto one  line for it to take effect.